core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -1464,8 +1495,8 @@ public void addTypesFromBundleBom(String yaml, ManagedBundle bundle, boolean for
     }
     
     @Override @Beta
-    public Collection<Throwable> validateType(RegisteredType typeToValidate) {
-        ReferenceWithError<RegisteredType> result = resolve(typeToValidate);
+    public Collection<Throwable> validateType(RegisteredType typeToValidate, RegisteredTypeLoadingContext constraint) {
```

Comment:
that context object is used primarily to constrain how loading can occur, e.g. specifying required supertypes, already encountered types in type hierarchy (to prevent loop), classpath, etc, that's why